From: Daniel van Vugt <daniel.van.vugt@canonical.com>
Date: Wed, 6 Mar 2019 14:48:10 +0000
Subject: js/ui/background.js: Use Ubuntu purple to match login screen

Original author: Didier Roche <didrocks@ubuntu.com>
Bug-Ubuntu: https://launchpad.net/bugs/1813119
Last-Update: 2019-03-01
Forwarded: not-needed

This way the login animation will appear to expand over the login
screen (system background) instead of suddenly replacing it.
Original author: Didier Roche <didrocks@ubuntu.com>
Author: Daniel van Vugt <daniel.van.vugt@canonical.com>
Bug-Ubuntu: https://launchpad.net/bugs/1813119
Last-Update: 2019-03-01
Forwarded: not-needed
---
 js/ui/background.js | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/js/ui/background.js b/js/ui/background.js
index babc1e4..ab1dba0 100644
--- a/js/ui/background.js
+++ b/js/ui/background.js
@@ -96,6 +96,7 @@
 const { Clutter, GDesktopEnums, Gio, GLib, GnomeDesktop, Meta } = imports.gi;
 const Signals = imports.signals;
 
+const Desktop = imports.misc.desktop;
 const LoginManager = imports.misc.loginManager;
 const Main = imports.ui.main;
 const Params = imports.misc.params;
@@ -499,6 +500,25 @@ let _systemBackground;
 
 var SystemBackground = class SystemBackground {
     constructor() {
+        // Yaru is default ubuntu theme, match GDM look
+        if (Desktop.is("ubuntu")) {
+            if (_systemBackground == null) {
+                _systemBackground = new Meta.Background({ meta_display: global.display });
+                let [, ubuntuColor] = Clutter.Color.from_string('#2C001E');
+                _systemBackground.set_color(ubuntuColor);
+            }
+            this.actor = new Meta.BackgroundActor({ meta_display: global.display,
+                                                    monitor: 0,
+                                                    background: _systemBackground });
+            let id = GLib.idle_add(GLib.PRIORITY_DEFAULT, () => {
+                this.emit('loaded');
+                return GLib.SOURCE_REMOVE;
+            });
+            GLib.Source.set_name_by_id(id, '[gnome-shell] SystemBackground.loaded');
+
+            return;
+        }
+
         let file = Gio.File.new_for_uri('resource:///org/gnome/shell/theme/noise-texture.png');
 
         if (_systemBackground == null) {
